packages/fast-colors/src/colorlib/colorPalette.ts

```diff
@@ -0,0 +1,271 @@
+// tslint:disable:member-ordering
+// tslint:disable:prefer-for-of
+// tslint:disable:interface-name
+
+import { blendMultiply, blendOverlay, saturateViaLCH } from "./colorBlending";
+import { rgbToHSL } from "./colorConverters";
+import { ColorHSL } from "./colorHSL";
+import { ColorInterpolationSpace, interpolateByColorSpace } from "./colorInterpolation";
+import { ColorRGBA64 } from "./colorRGBA64";
+import { ColorScale } from "./colorScale";
+
+export interface IColorPaletteConfig {
+    baseColor?: ColorRGBA64;
+    steps?: number;
+    interpolationMode?: ColorInterpolationSpace;
+    scaleColorLight?: ColorRGBA64;
+    scaleColorDark?: ColorRGBA64;
+    clipLight?: number;
+    clipDark?: number;
+    saturationAdjustmentCutoff?: number;
+    saturationLight?: number;
+    saturationDark?: number;
+    overlayLight?: number;
+    overlayDark?: number;
+    multiplyLight?: number;
+    multiplyDark?: number;
+}
+
+export class ColorPalette {
+    public static readonly defaultPaletteConfig: IColorPaletteConfig = {
+        baseColor: new ColorRGBA64(0.5, 0.5, 0.5, 1),
+        steps: 7,
+        interpolationMode: ColorInterpolationSpace.RGB,
+        scaleColorLight: new ColorRGBA64(1, 1, 1, 1),
+        scaleColorDark: new ColorRGBA64(0, 0, 0, 1),
+        clipLight: 0.185,
+        clipDark: 0.16,
+        saturationAdjustmentCutoff: 0.05,
+        saturationLight: 0.35,
+        saturationDark: 1.25,
+        overlayLight: 0,
+        overlayDark: 0.25,
+        multiplyLight: 0,
+        multiplyDark: 0,
+    };
+
+    constructor(config: IColorPaletteConfig) {
+        this.config = {
+            baseColor: config.baseColor
+                ? config.baseColor
+                : ColorPalette.defaultPaletteConfig.baseColor,

```

Comment:
nit: a slightly shorter way of doing all of these would just be to use `||` so `baseColor: config.baseColor || ColorPalette.defaultPaletteConfig.baseColor`, etc.

Reply:
I think you can just do `this.config = Object.assign({}. ColorPalette.defaultPaletteConfig, config)` to make this even more simple. That will apply the static defaults and then clobber thosse values with config values if they exist.